### PR TITLE
Homework Podstawy C++ password-check

### PR DESCRIPTION
**Pull request by NikczemnyOkularnik**
Autors: @Kasia9311 @NikczemnyOkularnik

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Kasia9311

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @NikczemnyOkularnik

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @Kasia9311

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Kasia9311 for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @NikczemnyOkularnik

**Comment by ziobron** [comment]
⏱ 3 XP extra for @NikczemnyOkularnik for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @Kasia9311

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @NikczemnyOkularnik

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,61 @@
...
+    case ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter:
+        return "Password needs to have at least one special character";
+    case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter:
+        return "Password needs to have at least one uppercase letter";
+    case ErrorCode::PasswordsDoNotMatch:
+        return "Passwords do not match";
+    default:
+        return "out of control";
+    }
+}
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:26]: Add new line between functions

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,61 @@
...
+bool doPasswordsMatch(std::string passwordOne, std::string passwordTwo)
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:27]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,61 @@
...
+{
+    bool check = (passwordOne == passwordTwo);
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:29]: `()` is redundant.

Just `return lhs == rhs`

File: homework/password-check/validation.cpp [hunk 4]
```diff
@@ -1,2 +1,61 @@
...
+    return check;
+}
+
+ErrorCode checkPasswordRules(std::string passwordOne)
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:33]: Pass by const refrence. Try always avoid copy.

File: homework/password-check/validation.cpp [hunk 5]
```diff
@@ -1,2 +1,61 @@
...
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    }
+    if (std::none_of(passwordOne.begin(), passwordOne.end(), isupper))
+    {
+        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
+    }
+    return ErrorCode::Ok;
+}
+
+ErrorCode checkPassword(std::string passwordOne, std::string passwordTwo)
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:54]: Pass by const refrence. Try always avoid copy.

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**